src/main/java/com/minecolonies/coremod/items/ItemScepterPermission.java

```diff
@@ -0,0 +1,195 @@
+     * @param hitY the y coordinate
+     * @param hitZ the z coordinate
+     * @return the result
+     */
+    @Override
+    @NotNull
+    public EnumActionResult onItemUse(
+            ItemStack scepter,
+            EntityPlayer playerIn,
+            World worldIn,
+            BlockPos pos,
+            EnumHand hand,
+            EnumFacing facing,
+            float hitX,
+            float hitY,
+            float hitZ)
+    {
+        if (!scepter.hasTagCompound())
+        {
+            scepter.setTagCompound(new NBTTagCompound());
+        }
+        final NBTTagCompound compound = scepter.getTagCompound();
+
+        if (worldIn.isRemote)
+        {
+            return EnumActionResult.SUCCESS;
+        }
+
+        final ColonyView colonyView = ColonyManager.getClosestColonyView(worldIn, pos);
+        if (colonyView == null)
+        {
+            return EnumActionResult.FAIL;
+        }
+
+        return handleItemAction(compound, playerIn, worldIn, pos, colonyView);
+    }
+
+    /**
+     * Handles mid air use.
+     *
+     * @param scepter the item stack
+     * @param worldIn the world
+     * @param playerIn the player
+     * @param hand the hand
+     * @return the result
+     */
+    @Override
+    @NotNull
+    public ActionResult<ItemStack> onItemRightClick(
+            @NotNull ItemStack scepter,
+            World worldIn,
+            EntityPlayer playerIn,
+            EnumHand hand)
+    {
+        if (worldIn.isRemote)
+        {
+            return new ActionResult(EnumActionResult.PASS, scepter);
+        }
+
+        if (!scepter.hasTagCompound())
+        {
+            scepter.setTagCompound(new NBTTagCompound());
+        }
+        final NBTTagCompound compound = scepter.getTagCompound();
+
+        toggleItemMode(playerIn, compound);
+
+        return new ActionResult(EnumActionResult.SUCCESS, scepter);
+    }
+
+    @NotNull
+    private EnumActionResult handleItemAction(
+            NBTTagCompound compound,
+            EntityPlayer playerIn,
+            World worldIn,
+            BlockPos pos,
+            ColonyView colonyView)
+    {
+        String tagItemMode = compound.getString(TAG_ITEM_MODE);
+
+        switch(tagItemMode) {
+            case TAG_VALUE_MODE_BLOCK:
+                return handleAddBlockType(playerIn, worldIn, pos, colonyView);
+            case TAG_VALUE_MODE_LOCATION:
+                return handleAddLocation(playerIn, worldIn, pos, colonyView);
+            default:
+                toggleItemMode(playerIn, compound);
+                return handleItemAction(compound, playerIn, worldIn, pos, colonyView);
+        }
+    }
+
+    @NotNull
+    static private void toggleItemMode(EntityPlayer playerIn, NBTTagCompound compound)
+    {
+        String itemMode = compound.getString(TAG_ITEM_MODE);
+
+        switch (itemMode)
+        {
+            case TAG_VALUE_MODE_BLOCK:
+                compound.setString(TAG_ITEM_MODE, TAG_VALUE_MODE_LOCATION);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Make this line start at column 9. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AIndentationCheck)